pkg/liquidity-mining/test/DistributionScheduler.test.ts

```diff
@@ -0,0 +1,394 @@
+      const tx = await distributionScheduler.startDistributionForToken(
+        rewardTokenDistributor.address,
+        rewardToken.address
+      );
+      return tx.wait();
+    }
+
+    sharedBeforeEach('set reward token distributor to DistributionScheduler', async () => {
+      await rewardTokenDistributor.add_reward(rewardToken.address, distributionScheduler.address);
+      startTime = roundUpTimestamp(await currentTimestamp());
+      for (let i = 0; i < 10; i++) {
+        await scheduleDistribution(100, startTime.add(i * WEEK));
+      }
+      await advanceToTimestamp(startTime.add(1));
+    });
+
+    it('transfers tokens to the gauge', async () => {
+      const pendingRewards = await distributionScheduler.getPendingRewards(
+        rewardTokenDistributor.address,
+        rewardToken.address
+      );
+
+      const receipt = await startDistributionForToken();
+
+      expectEvent.inIndirectReceipt(receipt, rewardToken.instance.interface, 'Transfer', {
+        from: distributionScheduler.address,
+        to: rewardTokenDistributor.address,
+        value: pendingRewards,
+      });
+    });
+
+    it('resets pending rewards to zero', async () => {
+      await startDistributionForToken();
+
+      const pendingRewards = await distributionScheduler.getPendingRewards(
+        rewardTokenDistributor.address,
+        rewardToken.address
+      );
+
+      expect(pendingRewards).to.be.eq(0);
+    });
+
+    context('when only some of the scheduled distributions are processed', () => {
+      it('updates the head node to point towards the first unprocessed node', async () => {
+        expect(await getNextNodeKey(HEAD)).to.be.eq(startTime);
+
+        await startDistributionForToken();
+
+        expect(await getNextNodeKey(HEAD)).to.be.eq(startTime.add(WEEK));
+      });
+    });
+
+    context('when the last scheduled distribution is processed', () => {
+      sharedBeforeEach('advance time past the last scheduled distribution', async () => {
+        await advanceToTimestamp(startTime.add(10 * WEEK));
+      });
+
+      it('updates the head node to point towards NULL', async () => {
+        expect(await getNextNodeKey(HEAD)).to.be.eq(startTime);
+
+        await startDistributionForToken();
+
+        expect(await getNextNodeKey(HEAD)).to.be.eq(NULL);
+      });
+    });
+  });
+
+  describe('startDistributions', () => {
+    let rewardToken2: Token;
+    let startTime: BigNumber;
+
+    sharedBeforeEach('deploy Token', async () => {
+      rewardToken2 = await Token.create('REWARD2');
+      await rewardToken2.mint(caller);
+      await rewardToken2.approve(distributionScheduler, MAX_UINT256, { from: caller });
+    });
+
+    sharedBeforeEach('initialize scheduled distributions', async () => {
+      await rewardTokenDistributor.add_reward(rewardToken.address, distributionScheduler.address);
+      await rewardTokenDistributor.add_reward(rewardToken2.address, distributionScheduler.address);
```

Comment:
Let's add a third reward with another distributor and check that it is correctly skipped